### PR TITLE
test(NODE-2422): Validate that mongocryptd is not spawned if bypassAutoEncryption=true

### PR DESCRIPTION
**Pull request by W-A-James**
#### What is changing?

- implement spec tests for bypassing spawning mongocryptd

##### Is there new documentation needed for these changes?

- no

#### What is the motivation for this change?

Spec compliance

### Double check the following

- [X] Ran `npm run check:lint` script
- [X] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [X] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [X] Changes are covered by tests
- [ ] New TODOs have a related JIRA ticket

### CODE REVIEW

File: test/integration/client-side-encryption/client_side_encryption.prose.test.js [hunk 1]
```diff
@@ -1103,13 +1103,178 @@ describe('Client Side Encryption Prose Tests', metadata, function () {
...
+        expect(insertError).to.be.instanceOf(
+          MongoServerSelectionError,
+          `
+
+TODO(NODE-5283): The error thrown in this test fails an instanceof check with MongoServerSelectionError.
+  This should change after NODE-5283.  If this assertion is failing, then the test
+  should be updated to reflect that the error thrown is now a server selection error.
+
+`
+        );
```
Comment (baileympearson) [inline test/integration/client-side-encryption/client_side_encryption.prose.test.js:1192]: This assertion is not necessary, but will fail once NODE-5283 is completed and will remind us to update this test.

**Review by baileympearson — APPROVED** [review]

**Review by durran — APPROVED** [review]

File: test/integration/client-side-encryption/client_side_encryption.prose.test.js (hunk 1 above)
Comment (dariakp) [reply test/integration/client-side-encryption/client_side_encryption.prose.test.js:1192]: so should it be inverted above? `not.to.be.instanceOf`?

**Review by dariakp — CHANGES_REQUESTED** [review]
There are a lot of failing tests on this PR specifically, have they all been reviewed?

**Review by dariakp — APPROVED** [review]
LGTM assuming all the CI failures are accounted for

**Comment by durran** [comment]
Win/Node 14 issues unrelated to this PR.

**Pull request merged**